forms-web-app/src/routes/eligibility.js

```diff
@@ -35,7 +33,7 @@ router.get('/listed-building', listedBuildingController.getListedBuilding);
 router.post(
   '/listed-building',
   listedBuildingValidationRules(),
-  listedBuildingValidator,
```

Comment:
So validator is the one by defaults with common validation result and each service with his own validator. Are we good ?

Reply:
Yes, I was following the pattern already set - I think by Jon Morgan, but maybe wrong.

The ordering of these 'two' middleware is important. 

The rules are specific to the given endpoint. There are `listedBuildingValidationRules()` and `decisionDateValidationRules()` and `yourDetailsRules()`.

Each function returns one or more express validator validation middlewares - https://express-validator.github.io/docs/check-api.html

All of these are run against the `req` and passed on to the next middleware.

Finally, the `validator` is common - and really not a validator at all, so maybe the name is bad and misleading - but is concerned with taking any validation errors from the previous middlewares (via examining the `req` object) and then adding the errors and error summary as properties on the `req.body` for display in the template.

The (badly named) `validator` is the same for every route so was extracted out to a separate file as previously it was being duplicated in each validation process.

Reply:
`validator` has been renamed to `validation-error-handler` to better represent what is actually happening.